tests/onnx/quantization/test_ptq_params.py

```diff
@@ -84,17 +94,20 @@ def test_params(self):
                 "nncf_graph": NNCFGraphToTest(
                     ONNXConvolutionMetatype, ONNXExtendedLayerAttributes(None, None)
                 ).nncf_graph,
-                "pattern": GraphPattern(),
+                "hw_patterns": get_hw_patterns(),
+                "ignored_patterns": get_ignored_pattens(),
             },
             "test_ignored_scopes": {
                 "nncf_graph": NNCFGraphToTest(
                     ONNXConvolutionMetatype, ONNXExtendedLayerAttributes(None, None)
                 ).nncf_graph,
-                "pattern": GraphPattern(),
+                "hw_patterns": get_hw_patterns(),
+                "ignored_patterns": get_ignored_pattens(),
             },
             "test_model_type_pass": {
                 "nncf_graph": NNCFGraphToTestMatMul(ONNXLinearMetatype).nncf_graph,
-                "pattern": GraphPattern(),
+                "hw_patterns": get_hw_patterns(),
+                "ignored_patterns": get_ignored_pattens(),
             },
         }
```

Comment:
Refactor this to use properly typed dataclasses with properly typed fields instead of opaque dicts.

Reply:
What's wrong with dicts?

Reply:
Dict accesses cannot be easily refactored with IDE support. For a dict, an IDE cannot look up whether a field exists at all or not, in order to catch an error at static analysis time. It is hard to set up proper type hints for a dict with varying key and value types. 

Dicts should only be used for storing collections while utilizing the mapping and algo asymptotics of the hash table. In other cases, you should set up a properly named and typed structure (a.k.a. class, or dataclass)